devito/ir/support/space.py

```diff
@@ -223,6 +241,17 @@ def drop(self, d):
     def negate(self):
         return self._construct([i.negate() for i in self.intervals])
 
+    def __getitem__(self, dim):
+        if not self.is_well_defined:
+            raise ValueError("Cannot fetch Interval from ill defined Space")
+        for i in self.intervals:
+            if i.dim == dim:
+                return i
+
+
+class DataSpace(Space):
+    pass
```

Comment:
What is this class for if empty?

Reply:
So that code becomes self-explanatory when used

Reply:
Not sure I understand, but trust you on this.

Reply:
Basically, using appropriate names at the right place helps people unfamiliar with the code understanding what's going on. Here I thought using ``DataSpace(...)`` instead of ``Space(...)`` was better in terms of self-documented code. Just a minor thing though.